rust-peer/src/main.rs

```diff
@@ -199,6 +224,39 @@ async fn main() -> Result<()> {
                 SwarmEvent::Behaviour(BehaviourEvent::Kademlia(e)) => {
                     debug!("Kademlia event: {:?}", e);
                 }
+                SwarmEvent::Behaviour(BehaviourEvent::RequestResponse(
+                    request_response::Event::Message { message, .. },
+                )) => match message {
+                    request_response::Message::Request { request, .. } => {
+                        todo!("request_response::Message::Request: {:?}", request);
+                    }
+                    request_response::Message::Response { response, .. } => {
+                        let file_body = response.0;
+                        info!(
+                            "request_response::Message::Response: size:{}",
+                            file_body.len()
+                        );
+                        // TODO: keep this file in memory and provider it via Kademlia
```

Comment:
Perhaps not in memory but on disk in a temporary directory? Any incoming request could then go into this temporary directory and serve the file. Kademlia could be used to discover, which peers ended up storing the file.

Reply:
Yeah. I just updated the comment: b286c4c. For now, I'd like to leave this as TODO because providing files in this rust-peer isn't very necessary in this application.